File: src/main/java/modules/music/AudioHandler.java
package modules.music;

import com.sedmelluq.discord.lavaplayer.player.AudioLoadResultHandler;
import com.sedmelluq.discord.lavaplayer.tools.FriendlyException;
import com.sedmelluq.discord.lavaplayer.track.AudioPlaylist;
import com.sedmelluq.discord.lavaplayer.track.AudioTrack;

import net.dv8tion.jda.api.entities.MessageChannel;
import utils.MessageUtils;

public class AudioHandler implements AudioLoadResultHandler {

    private GuildMusicManager musicManager;
    private MessageChannel channel;
    private String track;

    public AudioHandler(GuildMusicManager musicManager, MessageChannel channel, String track) {
	super();
	this.musicManager = musicManager;
	this.channel = channel;
	this.track = track;
    }

    @Override
    public void trackLoaded(AudioTrack track) {
	MessageUtils.sendMessage(channel, "Adding to queue " + track.getInfo().title);
	musicManager.scheduler.queue(track);
    }

    @Override
    public void playlistLoaded(AudioPlaylist playlist) {
	AudioTrack track = playlist.getSelectedTrack();

	if (track != null) {
	    // Users picked a track from a playlist: only load this one
	    track = playlist.getTracks().get(0);
	    musicManager.scheduler.queue(track);
	    MessageUtils.sendMessage(channel, "Adding to queue " + track.getInfo().title + " (first track of playlist "
		    + playlist.getName() + ")");
	} else {
	    // Load all the tracks from the playlist
	    for (AudioTrack t : playlist.getTracks())
		musicManager.scheduler.queue(t);

	    MessageUtils.sendMessage(channel, "Adding to queue all tracks from playlist " + playlist.getName());
	}
    }

    @Override
    public void noMatches() {
	MessageUtils.sendMessage(channel, "Nothing found by " + track);
    }

    @Override
    public void loadFailed(FriendlyException exception) {
	MessageUtils.sendMessage(channel, "Could not play: " + exception.getMessage());
    }

}

File: src/main/kotlin/com/github/sylux6/watanabot/commands/music/entities/AudioHandler.kt
package com.github.sylux6.watanabot.commands.music.entities

import com.sedmelluq.discord.lavaplayer.player.AudioLoadResultHandler
import com.sedmelluq.discord.lavaplayer.tools.FriendlyException
import com.sedmelluq.discord.lavaplayer.track.AudioPlaylist
import com.sedmelluq.discord.lavaplayer.track.AudioTrack
import net.dv8tion.jda.api.entities.MessageChannel
import com.github.sylux6.watanabot.utils.MessageUtils.sendMessage

class AudioHandler(private val musicManager: GuildMusicManager, private val channel: MessageChannel, private val track: String) : AudioLoadResultHandler {

    override fun trackLoaded(track: AudioTrack) {
        sendMessage(channel, "Adding to queue ${track.info.title}")
        musicManager.scheduler.queue(track)
    }

    override fun playlistLoaded(playlist: AudioPlaylist) {
        var track: AudioTrack? = playlist.selectedTrack

        if (track != null) {
            // Users picked a track from a playlist: only load this one
            track = playlist.tracks[0]
            musicManager.scheduler.queue(track)
            sendMessage(channel, "Adding to queue ${track!!.info.title} (first track of playlist ${playlist.name})")
        } else {
            // Load all the tracks from the playlist
            for (t in playlist.tracks)
                musicManager.scheduler.queue(t)

            sendMessage(channel, "Adding to queue all tracks from playlist ${playlist.name}")
        }
    }

    override fun noMatches() {
        sendMessage(channel, "Nothing found by $track")
    }

    override fun loadFailed(e: FriendlyException) {
        sendMessage(channel, "Could not play: ${e.message}")
    }

}
